### PR TITLE
User LDAP Groups

### PR DESCRIPTION
**Pull request by dlpierce**
Adds optional support for assigning user abilities by LDAP group.

### CODE REVIEW

**Review by aploshay — APPROVED** [review]
Looks good

File: app/models/ability.rb [hunk 1]
```diff
@@ -64,10 +64,16 @@ def curator_permissions
     curation_concern_read_permissions
   end
 
-  # Abilities that should be granted to patron
+  # Abilities that should be granted to patrons in an authorized group
+  # Ability to read authenticated visibility items is handled by #user_groups
+  def music_patron_permissions
```
Comment (mwoodiupui) [inline app/models/ability.rb:69]: Is there some way to move these site-specifics out of the model and into configuration?

File: app/models/ability.rb [hunk 2]
```diff
@@ -112,6 +131,6 @@ def curation_concerns
     end
 
     def roles
-      ['anonymous', 'campus_patron', 'curator', 'fulfiller', 'editor', 'image_editor', 'admin']
+      ['anonymous', 'music_patron', 'campus_patron', 'curator', 'fulfiller', 'editor', 'image_editor', 'admin']
```
Comment (mwoodiupui) [inline app/models/ability.rb:134]: Move this list (or, at least, the parts not those which must always be present) to configuration?

File: app/models/concerns/iu_user_roles.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-module PulUserRoles
```
Comment (mwoodiupui) [inline app/models/concerns/iu_user_roles.rb:1]: We need to find a way to configure-in site-specific modules and separate them from common code.

File: app/models/user.rb [hunk 4]
```diff
@@ -9,7 +9,11 @@ class User < ActiveRecord::Base
   # Connects this user object to Role-management behaviors.
   include Hydra::RoleManagement::UserRoles
   # include Sufia::UserUsageStats
-  include ::PulUserRoles
+  include ::IuUserRoles
```
Comment (mwoodiupui) [inline app/models/user.rb:12]: This should be configurable somehow.

**Review by mwoodiupui — APPROVED** [review]
I would have liked to see the site specifics separated out more, but otherwise looks reasonable.

**Pull request merged**